### PR TITLE
Handle libraries with call protection

### PR DESCRIPTION
**Pull request by marcocastignoli**
See #996

### CODE REVIEW

**Comment by marcocastignoli** [comment]
Since we discovered that this PR should handle all the cases, can we proceed? From my pov after writing a test we can merge, what do you think @kuzdogan?

**Comment by kuzdogan** [comment]
Does this also cover the `via-IR` case? If so yes, let's add 
1. legacy compilation placeholder test case
2.`via-IR` test case 

and merge 👍

**Comment by marcocastignoli** [comment]
> Does this also cover the via-IR case? If so yes, let's add

I suppose the actual function `replaceImmutableReferences` already does this.

**Comment by marcocastignoli** [comment]
I added the tests

File: packages/lib-sourcify/src/lib/verification.ts [hunk 1]
```diff
@@ -506,6 +513,20 @@ export function addLibraryAddresses(
...
+  const push20CodeOp = '73';
+  const callProtection = `0x${push20CodeOp}${'00'.repeat(20)}`;
+
+  if (template.startsWith(callProtection)) {
+    const replacedCallProtection = real.slice(0, 0 + callProtection.length);
+    return replacedCallProtection + template.substring(callProtection.length);
+  }
+  return template;
+}
+
```
Comment (kuzdogan) [inline packages/lib-sourcify/src/lib/verification.ts:529]: Doesn't this code only check the beginning of the bytecode with `.startsWith()`? 

How is the `viaIR` case passing? Does the `viaIR` case here have `0x73` at the beginning of the deployed bytecode? We should handle the case when the call protection is found in the middle of the bytecode.

**Review by kuzdogan — CHANGES_REQUESTED** [review]

File: packages/lib-sourcify/src/lib/verification.ts (hunk 1 above)
Comment (marcocastignoli) [reply packages/lib-sourcify/src/lib/verification.ts:529]: no, it handles it via the `immutableReferences`.

Comment (kuzdogan) [reply packages/lib-sourcify/src/lib/verification.ts:529]: Ahh right

**Review by kuzdogan — APPROVED** [review]

**Pull request merged**